src/geh_common/databricks/databricks_api_client.py

```diff
@@ -37,6 +54,17 @@ def start_job(self, job_id: int, python_params: list[str]) -> int:
         response = self.client.jobs.run_now(job_id=job_id, python_params=python_params)
         return response.run_id
 
+    def cancel_job_run(self, job_run_id: int, wait_for_cancellation: bool = True) -> None:
+        """Stop a Databricks job.
+
+        Args:
+            job_run_id (int): The ID of the job run.
+            wait_for_cancellation (bool, optional): Whether to wait for the job to be canceled. Defaults to True.
+        """
+        self.client.jobs.cancel_run(run_id=job_run_id)
+        if wait_for_cancellation:
+            self.wait_for_job_state(run_id=job_run_id, target_states=[RunLifeCycleState.TERMINATED.value])
```

Comment:
Is it possible that the job instance will terminate with another state? E.g. FAILED?